client/authority-discovery/src/lib.rs

```diff
@@ -412,17 +435,17 @@ where
 	// one for the upcoming session. In addition it could be participating in the current authority
 	// set with two keys. The function does not return all of the local authority discovery public
 	// keys, but only the ones intersecting with the current authority set.
-	fn get_own_public_keys_within_authority_set(&mut self) -> Result<HashSet<AuthorityId>> {
-		let local_pub_keys = self.key_store
-			.read()
+	fn get_own_public_keys_within_authority_set(
+		key_store: &BareCryptoStorePtr,
+		client: &Client,
+	) -> Result<HashSet<AuthorityId>> {
```

Comment:
If we just kept this method as is and made it return an empty `HashSet` instead for sentries, couldn't we keep the rest of the logic above unchanged? I guess we could still keep the early exit in the method above.

Reply:
I am not sure returning an empty `HashSet` for sentry nodes would be more expressive. This reminds me of *C* code returning `-1` on error.

Do you think the changes above are intrusive?